Gems/PhysX/Code/Editor/DebugDraw.cpp

```diff
@@ -687,6 +687,30 @@ namespace PhysX
             [[maybe_unused]] const AZ::Vector3& colliderScale,
             [[maybe_unused]] const bool forceUniformScaling) const
         {
+            const float minXBounds = -(heightfieldShapeConfig.GetNumColumns() * heightfieldShapeConfig.GetGridResolution().GetX()) / 2.0f;
+            const float minYBounds = -(heightfieldShapeConfig.GetNumRows() * heightfieldShapeConfig.GetGridResolution().GetY()) / 2.0f;
+
+            for (int xIndex = 0; xIndex < heightfieldShapeConfig.GetNumColumns() - 1; xIndex++)
+            {
+                for (int yIndex = 0; yIndex < heightfieldShapeConfig.GetNumRows() - 1; yIndex++)
+                {
+                    const int index0 = yIndex * heightfieldShapeConfig.GetNumColumns() + xIndex;
+                    const int index1 = yIndex * heightfieldShapeConfig.GetNumColumns() + xIndex + 1;
+                    const int index2 = (yIndex + 1) * heightfieldShapeConfig.GetNumColumns() + xIndex + 1;
+                    const int index3 = (yIndex + 1) * heightfieldShapeConfig.GetNumColumns() + xIndex;
+
+                    const float x0 = minXBounds + heightfieldShapeConfig.GetGridResolution().GetX() * xIndex;
+                    const float x1 = minXBounds + heightfieldShapeConfig.GetGridResolution().GetX() * (xIndex + 1);
+                    const float y0 = minYBounds + heightfieldShapeConfig.GetGridResolution().GetY() * (yIndex + 1);
+                    const float y1 = minYBounds + heightfieldShapeConfig.GetGridResolution().GetY() * yIndex;
+                    
+                    debugDisplay.DrawWireQuad(
+                        AZ::Vector3(x0, y0, heightfieldShapeConfig.GetSamples()[index0].m_height),
+                        AZ::Vector3(x1, y0, heightfieldShapeConfig.GetSamples()[index1].m_height),
+                        AZ::Vector3(x1, y1, heightfieldShapeConfig.GetSamples()[index2].m_height),
+                        AZ::Vector3(x0, y1, heightfieldShapeConfig.GetSamples()[index3].m_height));
+                }
+            }
```

Comment:
`heightfieldShapeConfig.GetNumColumns()`, `heightfieldShapeConfig.GetGridResolution()` and `heightfieldShapeConfig.GetSamples()` are called `(numColums*numRows)*4` times inside the loop and they return the same data, they do not depend on the loop. So this can benefit from calling them only once outside the loop, putting them in const ref variables and avoid all those function calls.

Reply:
+1

Reply:
Won't drawing a wire quad at every step actually double the number of lines drawn since each quad shares and edge with a neighbor? This would have a significant performance cost. It would be better to draw only two lines (like top and left) for each quad, then at the end of the loop close the bottom / right of the final quads.

Reply:
Good point, done.

Reply:
Done